### PR TITLE
Make metadata service IP part of the seed configuration

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
Remove the metadata service IP method from the cloud botanist interface and make it part of the `Seed` configuration.

**Special notes for your reviewer**:
As discussed with @mvladev 

**Release note**:

```action operator
The `Seed` resource does now contain a new `.spec.blockCIDRs` field. Every CIDR that is added here will be blocked for communcation of any control plane component of shoots in this seed. Typically, you should add the (cloud) provider's metadata service CIDR(s) here. For shooted seed clusters you might want to extend your `shoot.garden.sapcloud.io/use-as-seed` annotation with `blockCIDRs`, e.g. `shoot.garden.sapcloud.io/use-as-seed="true,blockCIDRs=1.2.3.4/5;6.7.8.9/10`.
```

### CODE REVIEW

File: charts/global-network-policies/templates/allow-to-metadata.yaml [hunk 1]
```diff
@@ -11,11 +11,13 @@ spec:
...
     matchLabels:
       networking.gardener.cloud/to-metadata: allowed
-{{- if .Values.metadataService }}
+{{- if .Values.blockedAddresses }}
   egress:
+{{- range $i, $address := .Values.blockedAddresses }}
   - to:
     - ipBlock:
-        cidr: {{ .Values.metadataService }}
+        cidr: {{ $address }}
```
Comment (mvladev) [inline charts/global-network-policies/templates/allow-to-metadata.yaml:19]: The problem with this change is that it doesn't do what the `NetworkPolicy`'s name and description says it does.

**Review by mvladev — DISMISSED** [review]
/lgtm

File: charts/global-network-policies/templates/allow-to-metadata.yaml [hunk 2]
```diff
@@ -3,22 +3,29 @@ kind: NetworkPolicy
 metadata:
   annotations:
     gardener.cloud/description: |
-      Allows Egress from pods labeled with 'networking.gardener.cloud/to-metadata=allowed'
+      Allows Egress from pods labeled with 'networking.gardener.cloud/to-blocked-cidrs=allowed'
       to CloudProvider's specific metadata service IP.
-  name: allow-to-metadata
+  name: allow-to-blocked-cidrs
```
Comment (mvladev) [inline charts/global-network-policies/templates/allow-to-metadata.yaml:8]: This file should also be renamed?

File: charts/global-network-policies/templates/allow-to-metadata.yaml [hunk 3]
```diff
@@ -3,22 +3,29 @@ kind: NetworkPolicy
...
-        cidr: {{ .Values.metadataService }}
+        cidr: {{ $address }}
+{{- end }}
 {{- else }}
   egress: []
 {{- end }}
   policyTypes:
   - Egress
   ingress: []
+ss: []
```
Comment (mvladev) [inline charts/global-network-policies/templates/allow-to-metadata.yaml]: This is broken?

Comment (rfranzke) [reply charts/global-network-policies/templates/allow-to-metadata.yaml]: nice catch, fixed

File: charts/global-network-policies/templates/allow-to-metadata.yaml (hunk 2 above)
Comment (rfranzke) [reply charts/global-network-policies/templates/allow-to-metadata.yaml:8]: done

File: charts/global-network-policies/templates/allow-to-blocked-cidrs.yaml [hunk 4]
```diff
@@ -3,22 +3,25 @@ kind: NetworkPolicy
...
-        cidr: {{ .Values.metadataService }}
+        cidr: {{ $address }}
+{{- end }}
 {{- else }}
   egress: []
 {{- end }}
   policyTypes:
   - Egress
   ingress: []
+{{- end }}
```
Comment (mvladev) [inline charts/global-network-policies/templates/allow-to-blocked-cidrs.yaml]: extra end

**Review by mvladev — APPROVED** [review]

**Pull request merged**